### PR TITLE
Fixed tables list/edition

### LINKED ISSUE #7089: [Vaults UI] Have the "Create a table" button to work

### LINKED ISSUE #7112: [Vaults UI] Add switch for tables/documents in folder

### PR DESCRIPTION
**Pull request by tdraier**
fixes : https://github.com/dust-tt/dust/issues/7089 
fixes : https://github.com/dust-tt/dust/issues/7112

## Description

- Add switch in list between documents/tables

Screenshot 2024-09-04 at 15 29 06

- Allows to create table
- Fixed document load : documentId must be encoded
- Add missing endpoint for viewing table from data source view
- Removed duplicate useDocument hook

## Deploy Plan

deploy `front`

### CODE REVIEW

File: front/components/vaults/VaultDataSourceViewContentList.tsx [hunk 1]
```diff
@@ -123,12 +134,35 @@ export const VaultDataSourceViewContentList = ({
           </>
         )}
         {isFolder(dataSourceView.dataSource) && (
-          <FoldersHeaderMenu
-            owner={owner}
-            vault={vault}
-            folder={dataSourceView.dataSource}
-            contentActionsRef={contentActionsRef}
-          />
+          <>
```
Comment (JulesBelveze) [inline front/components/vaults/VaultDataSourceViewContentList.tsx:137]: Could we display the item that is selected??

File: front/components/vaults/ContentActions.tsx [hunk 2]
```diff
@@ -13,8 +13,11 @@ import { DocumentOrTableUploadOrEditModal } from "@app/components/data_source/Do
 import { MultipleDocumentsUpload } from "@app/components/data_source/MultipleDocumentsUpload";
 import { isFolder, isWebsite } from "@app/lib/data_sources";
 
-type ContentActionKey =
-  | "DocumentOrTableUploadOrEditModal"
+export type ContentActionKey =
+  | "EditDocument"
+  | "EditTable"
+  | "CreateDocument"
+  | "CreateTable"
```
Comment (JulesBelveze) [inline front/components/vaults/ContentActions.tsx:20]: Do we really need 4 actions? Couldn't we deal with only 2? (DocumentUploadOrEdit and TableUploadOrEdit)

Comment (tdraier) [reply front/components/vaults/ContentActions.tsx:20]: not really needed indeed

File: front/components/vaults/ContentActions.tsx [hunk 3]
```diff
@@ -14,10 +14,8 @@ import { MultipleDocumentsUpload } from "@app/components/data_source/MultipleDoc
...
 export type ContentActionKey =
-  | "EditDocument"
-  | "EditTable"
-  | "CreateDocument"
-  | "CreateTable"
+  | "DocumentUploadOrEdit"
+  | "TableUploadOrEdit"
   | "MultipleDocumentsUpload"
   | "DocumentOrTableDeleteDialog"
   | "TableUploadOrEditModal";
```
Comment (JulesBelveze) [inline front/components/vaults/ContentActions.tsx:21]: `nit`: not sure if this is still being used? 🤔

**Review by JulesBelveze — APPROVED** [review]
LGTM 🔥

File: front/components/vaults/ContentActions.tsx (hunk 3 above)
Comment (tdraier) [reply front/components/vaults/ContentActions.tsx]: good catch 👍

**Pull request merged**